### PR TITLE
Document the new release process

### LINKED ISSUE #56: Document the new release process
Propose a new release process and update Release.md and Changelog.md to reflect it.

I can think on something like:

1. Open a PR that bumps version refs in the repo, edit Chanelog.md with notable changes
2. Once this PR is merged, add a tag on this commit/main branch
3. Release

Open questions;

1. Do we need release candidates at this stage of the project?

/cc @robscott @shaneutt

### PR DESCRIPTION
**Pull request by LiorLieberman**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR add docs about the new release process + CHANGELOG.md to track the project changes

**Which issue(s) this PR fixes**:

Fixes #56 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @robscott @shaneutt @levikobi

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -1,9 +1,43 @@
...
+`main` branch. Approval of the PR indicates community consensus for a new
+release.
+
+### Patch a release
+
+<!-- TODO(liorlieberman) -->
+
+### Release a MAJOR or MINOR release
+
+<!-- TODO(liorlieberman) -->
```
Comment (LiorLieberman) [inline RELEASE.md:32]: @robscott @shaneutt I am a bit confused how gw is managing this. Mostly around if and when we cut "release branches" 
Would appreciate your clarification/input here

Comment (robscott) [reply RELEASE.md:32]: We generally only cut release branches after the minor release tag is cut. With that said, given the mixup we made with v0.8.1 (https://github.com/kubernetes-sigs/gateway-api/issues/2411), I think one of our AIs will be to cut a release branch before releasing and prevent tags on the main branch.

File: RELEASE.md [hunk 2]
```diff
@@ -1,9 +1,43 @@
...
+
+```
+go install k8s.io/release/cmd/release-notes@latest
+export GITHUB_TOKEN=your_token_here
+release-notes --start-sha EXAMPLE_COMMIT --end-sha EXAMPLE_COMMIT --branch main --repo ingress2gateway --org kubernetes-sigs
+```
+
+This output will likely need to be reorganized and cleaned up a bit, but it
+provides a good starting point. Once you're satisfied with the changelog, create
+a PR. This must go through the regular PR review process and get merged into the
```
Comment (robscott) [inline RELEASE.md:22]: One nit: In Gateway API we've generally aimed for approval from at least a majority of maintainers before a release, so we have a bit of a higher bar for review + approval here.

File: RELEASE.md [hunk 3]
```diff
@@ -1,9 +1,43 @@
...
+
+### Release a RC release
+
+1. Open a PR with changes of the version references in the codebase.
+2. Include necessary changelog updates to CHANGELOG.md in this PR.
+4. Merge the PR
+5. Tag the release using the commit on main where the PR merged. This can be done using the git CLI `git tag -sa $VERSION`.
+6. Run `git push origin $VERSION`, this will trigger a github workflow that will create the release.
+7. Verify the [releases page](https://github.com/kubernetes-sigs/ingress2gateway/releases) to ensure that the release meets the expectations.
+<!-- TODO(liorlieberman) is this needed? -->
```
Comment (robscott) [inline RELEASE.md:42]: I don't think so, we haven't bothered with Gateway API, but we do usually announce releases in Slack. If it's a particularly notable release, sending an email to the sig-network mailing list might make sense.

**Review by robscott — COMMENTED** [review]
Thanks @LiorLieberman!

File: RELEASE.md (hunk 2 above)
Comment (LiorLieberman) [reply RELEASE.md:22]: we will do the same here, I do not think there is a need to mention it here. this part is the same wording copied from Gateway Release.md

File: RELEASE.md [hunk 4]
```diff
@@ -1,9 +1,60 @@
 # Release Process
 
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+## Overview
+
+The Ingress2Gateway Project is CLI project that helps with translating Ingress and provider related resources to [Gateway API](https://github.com/kubernetes-sigs/gateway-api) resources.
```
Comment (levikobi) [inline RELEASE.md:5]:
```suggestion
The Ingress2Gateway Project is a CLI project that helps translate Ingress and provider-related resources to [Gateway API](https://github.com/kubernetes-sigs/gateway-api) resources.
```

**Pull request merged**